sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceStrategy.scala

```diff
@@ -448,47 +448,54 @@ object DataSourceStrategy {
     }
   }
 
+  private def translateFilterValues(catalystValue: Any, dataType: DataType): Any = {
+    dataType match {
+      case DateType => DateConverter.toScala(catalystValue)
```

Comment:
what if the predicate is `col = array(date_value)`? Seems better to add a new boolean flag in `createToScalaConverter`

Reply:
done